File: minigit-5/ts/src/commands/ls-files.ts
// ls-files command - List indexed files

import { findRepoRoot } from '../utils';
import { readIndex } from '../index-file';

export function lsFiles(args: string[]): number {
  const repoRoot = findRepoRoot();
  if (!repoRoot) {
    console.error('fatal: not a minigit repository');
    return 1;
  }

  // Parse arguments
  let showStaged = false;

  for (const arg of args) {
    if (arg === '--staged' || arg === '-s') {
      showStaged = true;
    }
  }

  // Read index
  const index = readIndex(repoRoot);
  const entries = index.entries.slice().sort((a, b) => a.path.localeCompare(b.path));

  for (const entry of entries) {
    if (showStaged) {
      // Format: mode sha stage path
      const mode = entry.mode.toString(8).padStart(6, '0');
      const stage = 0; // Stage is always 0 for regular index entries
      console.log(`${mode} ${entry.sha} ${stage}\t${entry.path}`);
    } else {
      console.log(entry.path);
    }
  }

  return 0;
}

File: minigit-5/py/commands/ls_files.py
"""ls-files command - List indexed files."""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))

from utils import find_repo_root
from index import Index


def run(args: list[str]) -> int:
    """List files in the index."""
    repo_root = find_repo_root()
    if repo_root is None:
        print("fatal: not a minigit repository", file=sys.stderr)
        return 1

    # Parse arguments
    staged = '--staged' in args or '-s' in args
    cached = '--cached' in args

    index = Index.read(repo_root)

    for entry in index.get_all_entries():
        if staged or cached:
            # Format: mode sha stage path
            stage = 0  # We don't track merge stages
            print(f"{entry.mode:06o} {entry.sha} {stage}\t{entry.name}")
        else:
            print(entry.name)

    return 0
